### PR TITLE
llvm: add linux dependencies, set platform-specific flags

### PR DESCRIPTION
**Pull request by iMichka**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/llvm.rb [hunk 1]
```diff
@@ -137,13 +139,29 @@ def install
       -DCLANG_PYTHON_BINDINGS_VERSIONS=#{py_ver}
     ]
 
-    sdk = MacOS.sdk_path_if_needed
-    args << "-DDEFAULT_SYSROOT=#{sdk}" if sdk
+    on_macos do
+      args << "-DLLVM_LINK_LLVM_DYLIB=ON"
```
Comment (Bo98) [inline Formula/llvm.rb:143]: This isn't platform specific. It's what makes LLVM link to itself via shared libraries and makes a massive difference on bottle size.

**Comment by carlocab** [comment]
Since we're making changes to the formula, can we clean up this mess I left here last time:
```diff
diff --git a/Formula/llvm.rb b/Formula/llvm.rb
index ef9f2cb33f2..acd3715d9fd 100644
--- a/Formula/llvm.rb
+++ b/Formula/llvm.rb
@@ -91,9 +91,8 @@ class Llvm < Formula
       libunwind
     ]
 
-    python = Formula["python@3.9"]
-    py_ver = Language::Python.major_minor_version(python)
-    site_packages = Language::Python.site_packages(python).delete_prefix("lib/")
+    py_ver = Language::Python.major_minor_version("python3")
+    site_packages = Language::Python.site_packages("python3").delete_prefix("lib/")
 
     # Apple's libstdc++ is too old to build LLVM
     ENV.libcxx if ENV.compiler == :clang
```
That should resolve correctly since Homebrew Python should be in `PATH`. (Right, @Bo98?)

**Comment by Bo98** [comment]
Yes it should work.

**Comment by Bo98** [comment]
We also forbid mixed Python versions in the dependency tree so there shouldn't be any confusion there (which there was during the Python 3.8 migration).

File: Formula/llvm.rb (hunk 1 above)
Comment (carlocab) [reply Formula/llvm.rb:143]: Yep, you definitely want this flag: https://llvm.org/docs/BuildingADistribution.html#options-for-reducing-size

**Comment by carlocab** [comment]
I think you can also make this macOS-only:

https://github.com/Homebrew/homebrew-core/blob/c5208b816bf8992f841c11eb0ee8bcab507f5b94/Formula/llvm.rb#L135

**Comment by iMichka** [comment]
Done. I did not change `DLLDB_PYTHON_RELATIVE_PATH=libexec/#{site_packages}` because we would need to test that first on Linux.

**Comment by carlocab** [comment]
Tried doing `import lldb` with a non-Homebrew Python in a Docker container. No segfaults (which is what you'd get if you tried it on a Mac).

You may wish to give this a more thorough test, but I just thought you'd like to know.

**Review by Bo98 — APPROVED** [review]

**Review by carlocab — APPROVED** [review]

**Pull request merged**